internal/store/pod.go

```diff
@@ -40,7 +40,12 @@ var (
 	podStatusReasons           = []string{"Evicted", "NodeAffinity", "NodeLost", "Shutdown", "UnexpectedAdmissionError"}
 )
 
-func podMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generator.FamilyGenerator {
+func podMetricFamilies(injectPerSeriesMetadata bool, allowAnnotationsList []string, allowLabelsList []string) []generator.FamilyGenerator {
+	mc := &MetricConfig{
```

Comment:
Doing this instead of adding 3 arguments to each generator, I add one.   Perhaps, I suppose it makes it less "functional" as the behavior of the function is dictated by a complex configuration object.

I feel like it might be time to bite the bullet and just have a config object than growing the argument lists. This also makes it easier to add options in the future for future features / refactors.